### ISSUE TITLE
Seperating Releases Accordingly

### ISSUE THREAD

**Issue by GaryQ** [firefly-iii/firefly-iii#2719]
Hi,
Although I love and appreciate all the hard work you have and continue to put into Firefly III I'm having trouble keeping up with what's what as you seem to release everything as if it was a stable release without proper segregation. 
I've been trying to keep track since 4.8 but it's hard to tell when and what I can finally start using for real.

I personally think the standard method of labeling and numbering each release as Alpha x, Beta x and Stable accordingly would be a great addition and remove unnecessary confusion. 

I think it would also help us better help you in testing and debugging.
1 - Use Stable Release for keeping track of our $$$$ < Intentional $ used as humorous revenge for all the hard coded Euros everywhere :) > - without too many worries and headaches
2 - And at the same time, for those that want to test and play around with Alpha and Beta versions separately know exactly what stage of stability to expect when helping out.

Regards,
GaryQ 
P.S. Just my 2 cents Canadian ;)

**Comment by JC5** [comment]
The master branch is always stable, and the releases should be stable. If you have a specific problem, please let me know.

**Comment by GaryQ** [comment]
With all due respect you should eliminate bugs before considering a version release as stable. Bugs are normal but should be fixed before a release is tagged as stable. Example:

![Annotation 2019-10-22 104933](https://user-images.githubusercontent.com/3269694/67305816-96593080-f4bb-11e9-9381-b2ef1e6a02d6.JPG)

![Annotation 2019-10-22 104932](https://user-images.githubusercontent.com/3269694/67305833-9fe29880-f4bb-11e9-80c6-902fe645d6cd.JPG)

**Comment by JC5** [comment]
Sorry, shit happens. I don't do it on purpose.

**Comment by GaryQ** [comment]
I know it does. Been programming since 1984. 
I'm just trying to help by suggesting that Versions be split and tagged accordingly  and that bugs be squashed in the current version till stable to be tagged as such and that new options or changes be tested in a separate version not the current one: Segregation.

Example
(Version numbering is just an example not a rule)

4.8.0. ALPHA 1 - (Anything can is possible and can change be added so don't use in production)
4.8..0,1 - Beta 1 - (No new additions - Bug tracking time - Please test it out - be kind to me)
4.8.0.2 - Stable Release 
4.8.0.2.1 (Shit slips through unnoticed sometimes)

4.8.1.1 - Beta 1
... ... ...
4.8.1.5 - Beta 5
4.8.1.6 - Stable release

Both versions can be going at the same time. One stable and one work version max though. Other experimental next versions kept out of the release cycle in it's own repository branch or off the grid.

Additions and changes are limited to the newer ALPHA until BETA releases not the previous release. 
Previous release should technically only have bug fixes because as you said "shit happens"
Let the new shit from changes happen in the new beta version limiting adding more bugs in the previous.

Of course STABLE is a relative term meaning as close as possible to bug free but should be limited to  to the occasional "Oh my ! I can't believe that slipped by everyone for so long. Thanks for spotting that Dude!"

I just think it would make life easier, especially for you, as bugs found in a beta release don't need any emergency fix. Less "against the wall" pressure to squash! They are "use at your own risk and thanks for your feedback" versions 

But it's your project and I like it ,otherwise I would be looking elsewhere, and there's nothing out there that I found better to use as a base without writing from scratch and it's just my personal suggestion not to be seen or interpreted as negative criticism. Otherwise I would not bother trying to help. It's actually the first project I have found worth helping out on. So I hope you can see the major compliment there.

**Comment by JC5** [comment]
Thanks man, make no mistake. Your feedback is really appreciated :)

**Comment by GaryQ** [comment]
Your welcome. It's nice to know you can accept feedback. :)
I'll avoid flaming you on the bloody annoying default EURO and symbol LOL. 
I just made some changes not done yet... And the Euro symbol on the side menu (which hides automatically)  is a $ now  <:-)

I was a bit nervous to post any suggestions even the little "CVS" typo comment as I've seen too many people get literally bashed for making any little suggestions over the years on open source projects.

**Comment by GaryQ** [comment]
HAHAHA, yeah I understand and can imagine the fun. 
That's exactly why I removed my copy of Ff-III from my GITHUB account. I make my mods slowly and locally for me. I can't keep up with your changes so imagine if they were open to "comments"? How come he's at version 9.9.9 and you are only at version 8.8.8? … or the famous "NEED CODE FOR <insert whatever> PLEASE"

**Comment by GaryQ** [comment]
BTW -  @BobWs avatar inspired me to finally find one I would like to use couldn;t find a great png of "Guitar Stuart" but found a close second "Banana Stuart"

On the serious side, I finally found the GITHUB project that I remembered seeing as a great example to follow. Take a peek (if you haven't already lately) at how https://github.com/ColorlibHQ/AdminLTE is structured for multiple versions 2.4/x and the 3.0.0 in progress and having the different branch releases at the same time. 
1 - Check the releases page.
2 - Notice how he follows semver versioning to the letter. Uh no... Technically that's to the number.
3 - Also an important point he separates each changes or fix into individual commits. Sure it makes more commits but rolling back one doesn't affect the other. 

I noticed you added a versioning system to the dev branch. Love the info text! It's funny :)

Personally I'd resist and new breaking changes to the 4.8.x branch and limit it to bug fixes only and start a new clean branch maybe 5.0.0 ALPHA1 as a fresh start to this chapter in your development..
But that's just my C$ 0.02. as usual :)
